src/status_im/chat/events/sign_up.cljs

```diff
@@ -49,43 +49,80 @@
                            :message-status :seen}}))
 
 (handlers/register-handler-fx
-  ::sign-up-success
-  [re-frame/trim-v (re-frame/inject-cofx :random-id)]
-  (fn [{:keys [db random-id]} message-id]
-    (-> {:db db
-         :dispatch-n [;; create manual way for entering confirmation code
-                      (sign-up/enter-confirmation-code-event random-id)
-                      ;; create automatic way for receiving confirmation code
-                      sign-up/start-listening-confirmation-code-sms-event]}
-        (message-seen message-id))))
+ :start-listening-confirmation-code-sms
+ [re-frame/trim-v]
+ (fn [{:keys [db]} [sms-listener]]
+   {:db (if-not (:confirmation-code-sms-listener db)
+          (assoc db :confirmation-code-sms-listener sms-listener)
+          db)}))
+
+(defn stop-listening-confirmation-code-sms [{:keys [db] :as fx}]
+  (-> fx
+      (update-in [:db :confirmation-code-sms-listener] dissoc)
```

Comment:
I think this won't dissoc anything, it should be `(update :db dissoc :confirmation-code-sms-listener)`